bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/veneer/BigtableHBaseVeneerSettings.java

```diff
@@ -273,99 +289,106 @@ private void configureMetricsBridge(Builder settings) {
-
-    BigtableInstanceAdminStubSettings.Builder stubSettings = instanceAdminBuilder.stubSettings();
-
-    configureConnection(stubSettings, BIGTABLE_ADMIN_HOST_KEY);
-
-    configureHeaderProvider(stubSettings);
-
-    configureCredentialProvider(stubSettings);
-
-    return instanceAdminBuilder.build();
-  }
-
-  private void configureConnection(StubSettings.Builder stubSettings, String endpointKey) {
-
-    String endpoint = stubSettings.getEndpoint();
-    String portNumber = configuration.get(BIGTABLE_PORT_KEY);
-    if (isNullOrEmpty(portNumber)) {
-      portNumber = endpoint.substring(endpoint.lastIndexOf(":") + 1);
+  private void configureConnection(StubSettings.Builder<?, ?> stubSettings, String endpointKey) {
+    String defaultEndpoint = stubSettings.getEndpoint();
+    String defaultHostname = defaultEndpoint.substring(0, defaultEndpoint.lastIndexOf(':'));
+    String defaultPort = defaultEndpoint.substring(defaultEndpoint.lastIndexOf(':') + 1);
+
+    Optional<String> hostOverride = Optional.fromNullable(configuration.get(endpointKey));
+    Optional<String> portOverride = Optional.fromNullable(configuration.get(BIGTABLE_PORT_KEY));
+    Optional<String> endpointOverride = Optional.absent();
+
+    if (hostOverride.isPresent() || portOverride.isPresent()) {
+      endpointOverride =
+          Optional.of(hostOverride.or(defaultHostname) + ":" + portOverride.or(defaultPort));
+    } else if (endpointKey.equals(BIGTABLE_HOST_KEY)
+        && configuration.getBoolean(BIGTABLE_USE_BATCH, false)) {
+      endpointOverride = Optional.of(BIGTABLE_BATCH_DATA_HOST_DEFAULT + ":443");
     }
 
-    String hostOverride = configuration.get(endpointKey);
-    if (isNullOrEmpty(hostOverride)) {
-      hostOverride = endpoint.substring(0, endpoint.lastIndexOf(":"));
-      LOG.debug("%s is configured at %s", endpointKey, hostOverride);
+    if (endpointOverride.isPresent()) {
+      stubSettings.setEndpoint(endpointOverride.get());
+      LOG.debug("%s is configured at %s", endpointKey, endpointOverride);
     }
 
-    if (isBatchModeEnabled() && BIGTABLE_HOST_KEY.equals(endpointKey)) {
-      // TODO: move this constant in default alignment PR.
-      hostOverride = BIGTABLE_BATCH_DATA_HOST_DEFAULT;
-    }
-    stubSettings.setEndpoint(hostOverride + ":" + portNumber);
+    InstantiatingGrpcChannelProvider.Builder channelProvider =
+        ((InstantiatingGrpcChannelProvider) stubSettings.getTransportChannelProvider()).toBuilder();
 
-    InstantiatingGrpcChannelProvider.Builder channelBuilder = defaultGrpcTransportProviderBuilder();
-
-    if (Boolean.parseBoolean(configuration.get(BIGTABLE_USE_PLAINTEXT_NEGOTIATION))) {
-      channelBuilder.setChannelConfigurator(
+    if (configuration.getBoolean(BIGTABLE_USE_PLAINTEXT_NEGOTIATION, false)) {
+      @SuppressWarnings("rawtypes")
+      final ApiFunction<ManagedChannelBuilder, ManagedChannelBuilder> prevConfigurator =
```

Comment:
a little confused here - why are we switching from channelBuilder to channelProvider and why the previous configurator?